src/core/public/fatal_errors/fatal_errors_service.tsx

```diff
@@ -58,13 +58,23 @@ export interface FatalErrorsSetup {
 export class FatalErrorsService {
   private readonly errorInfo$ = new Rx.ReplaySubject<ErrorInfo>();
 
-  constructor(private rootDomElement: HTMLElement, private stopCoreSystem: () => void) {}
+  /**
+   *
+   * @param rootDomElement
+   * @param onFirstErrorCb - Callback function that gets executed after the first error,
+   *   but before the FatalErrorsService renders the error to the DOM. Used by Core to
+   *   to stop all services before showing the error page.
```

Comment:
**optional**: probably we shouldn't comment how the core service reacts on the callback. the comment can be outdated easily